File: source/cpBase/CPDocBaseClass.vb
'
' documentation should be in a new project that inherits these classes. The class names should be the object names in the actual cp project
'
Namespace Contensive.BaseClasses
    Public MustInherit Class CPDocBaseClass
        Public MustOverride Property Content() As String
        Public MustOverride ReadOnly Property NavigationStructure() As String
        Public MustOverride Property NoFollow() As Boolean
        Public MustOverride ReadOnly Property PageId() As Integer
        Public MustOverride ReadOnly Property PageName() As String
        Public MustOverride ReadOnly Property RefreshQueryString() As String
        Public MustOverride ReadOnly Property SectionId() As Integer
        Public MustOverride ReadOnly Property StartTime() As Date
        Public MustOverride ReadOnly Property TemplateId() As Integer
        Public MustOverride ReadOnly Property Type() As String
        Public MustOverride Sub AddHeadStyle(ByVal StyleSheet As String)
        Public MustOverride Sub AddHeadStyleLink(ByVal StyleSheetLink As String)
        Public MustOverride Sub AddHeadJavascript(ByVal NewCode As String)
        Public MustOverride Sub AddHeadTag(ByVal HeadTag As String)
        Public MustOverride Sub AddMetaDescription(ByVal MetaDescription As String)
        Public MustOverride Sub AddMetaKeywordList(ByVal MetaKeywordList As String)
        Public MustOverride Sub AddOnLoadJavascript(ByVal NewCode As String)
        Public MustOverride Sub AddTitle(ByVal PageTitle As String)
        Public MustOverride Sub AddRefreshQueryString(ByVal Name As String, ByVal Value As String)
        Public MustOverride Sub AddBodyEnd(ByVal NewCode As String)
        Public MustOverride Property Body() As String
        Public MustOverride ReadOnly Property SiteStylesheet() As String
        Public MustOverride Sub SetProperty(ByVal FieldName As String, ByVal FieldValue As String)
        Public MustOverride Function GetProperty(ByVal PropertyName As String, Optional ByVal DefaultValue As String = "") As String
        Public MustOverride Function GetText(ByVal PropertyName As String, Optional ByVal DefaultValue As String = "") As String
        Public MustOverride Function GetBoolean(ByVal PropertyName As String, Optional ByVal DefaultValue As String = "") As Boolean
        Public MustOverride Function GetDate(ByVal PropertyName As String, Optional ByVal DefaultValue As String = "") As Date
        Public MustOverride Function GetInteger(ByVal PropertyName As String, Optional ByVal DefaultValue As String = "") As Integer
        Public MustOverride Function GetNumber(ByVal PropertyName As String, Optional ByVal DefaultValue As String = "") As Double
        Public MustOverride Function IsProperty(ByVal PropertyName As String) As Boolean
        Public MustOverride ReadOnly Property IsAdminSite() As Boolean
        ' deprecated - replaced with Setproperty because c-sharp does not support properties with arguments
        '----------------------------------------------------------------------------------------------------
        'Public MustOverride Property GlobalVar(ByVal Index As String) As String
        Public MustOverride Function get_GlobalVar(ByVal Index As String) As String
        Public MustOverride Sub set_GlobalVar(ByVal Index As String, ByVal Value As String)
        '----------------------------------------------------------------------------------------------------
        'Public MustOverride ReadOnly Property IsGlobalVar(ByVal Index As String) As Boolean
        Public MustOverride Function get_IsGlobalVar(ByVal Index As String) As Boolean
        '----------------------------------------------------------------------------------------------------
        'Public MustOverride ReadOnly Property IsVar(ByVal Index As String) As Boolean
        Public MustOverride Function get_IsVar(ByVal Index As String) As Boolean
        '----------------------------------------------------------------------------------------------------
        'Public MustOverride Property Var(ByVal Index As String) As String
        Public MustOverride Function get_Var(ByVal Index As String) As String
        Public MustOverride Sub Var(ByVal Index As String, ByVal Value As String)

    End Class

End Namespace

File: source/CPBase50/CPDocBaseClass.cs






//
// documentation should be in a new project that inherits these classes. The class names should be the object names in the actual cp project
//


using System;
using System.Collections;
using System.Collections.Generic;
using System.Diagnostics;

namespace Contensive.BaseClasses
{
	public abstract class CPDocBaseClass
	{
		public abstract string Content {get; set;}
        [Obsolete("Use addon navigation.", true)] public abstract string NavigationStructure {get;}
		public abstract bool NoFollow {get; set;}
		public abstract int PageId {get;}
		public abstract string PageName {get;}
		public abstract string RefreshQueryString {get;}
        [Obsolete("Section is no longer supported", true)] public abstract int SectionId {get;}
		public abstract DateTime StartTime {get;}
		public abstract int TemplateId {get;}
		public abstract string Type {get;}
		public abstract void AddHeadStyle(string StyleSheet);
		public abstract void AddHeadStyleLink(string StyleSheetLink);
		public abstract void AddHeadJavascript(string NewCode);
		public abstract void AddHeadTag(string HeadTag);
		public abstract void AddMetaDescription(string MetaDescription);
		public abstract void AddMetaKeywordList(string MetaKeywordList);
		public abstract void AddOnLoadJavascript(string NewCode);
		public abstract void AddTitle(string PageTitle);
		public abstract void AddBodyEnd(string html);
		public abstract string Body {get; set;}
        [Obsolete("Site styles are no longer supported. Include styles and javascript in addons.", true)] public abstract string SiteStylesheet {get;}
		public abstract void SetProperty(string FieldName, string FieldValue);
		public abstract string GetProperty(string PropertyName, string DefaultValue = "");
		public abstract string GetText(string PropertyName, string DefaultValue = "");
		public abstract bool GetBoolean(string PropertyName, string DefaultValue = "");
		public abstract DateTime GetDate(string PropertyName, string DefaultValue = "");
		public abstract int GetInteger(string PropertyName, string DefaultValue = "");
		public abstract double GetNumber(string PropertyName, string DefaultValue = "");
		public abstract bool IsProperty(string PropertyName);
        public abstract bool IsAdminSite {get;}
        public abstract string get_GlobalVar(string Index);
        public abstract void set_GlobalVar(string Index, string Value);
        public abstract bool get_IsGlobalVar(string Index);
        public abstract bool get_IsVar(string Index);
        public abstract string get_Var(string Index);
        public abstract void set_Var(string Index, string Value);
        //
        public abstract void AddRefreshQueryString(string Name, string Value);
        public abstract void AddRefreshQueryString(string Name, int Value);
        public abstract void AddRefreshQueryString(string Name, Double Value);
        public abstract void AddRefreshQueryString(string Name, bool Value);
        public abstract void AddRefreshQueryString(string Name, DateTime Value);

    }
}

